e2e/test/iothub/method/MethodE2ETests.cs

```diff
@@ -142,18 +143,22 @@ public async Task Method_ServiceInvokeDeviceMethodWithUnknownDeviceThrows()
             };
 
             // act
-            ErrorCode actualErrorCode = ErrorCode.InvalidErrorCode;
+            HttpStatusCode errorCode = HttpStatusCode.OK;
+            IotHubStatusCode iotHubStatusCode = IotHubStatusCode.Unknown;
             try
             {
                 // Invoke the direct method asynchronously and get the response from the simulated device.
                 await serviceClient.DirectMethods.InvokeAsync("SomeNonExistantDevice", methodInvocation);
             }
-            catch (DeviceNotFoundException ex)
+            catch (IotHubServiceException ex) when (ex.ErrorCode is HttpStatusCode.NotFound)
             {
-                actualErrorCode = ex.Code;
+                errorCode = ex.ErrorCode;
+                iotHubStatusCode = ex.IotHubStatusCode;
             }
 
-            Assert.AreEqual(ErrorCode.DeviceNotFound, actualErrorCode);
+            // assert
+            Assert.AreEqual(HttpStatusCode.NotFound, errorCode);
+            Assert.AreEqual(IotHubStatusCode.DeviceNotFound, iotHubStatusCode);
```

Comment:
Please take the time to change these to fluent assertions:

```suggestion
            errorCode.Should().Be(HttpStatusCode.NotFound);
            iotHubStatusCode.Should().Be(IotHubStatusCode.DeviceNotFound);
```